### PR TITLE
[DRAFT] External link support

### PR DESCRIPTION
**Pull request by TomNUSDS**
- Make links work on both staging and production. Staging "is one level down" [https://usds.github.io/website-staging/](https://usds.github.io/website-staging/) versus production which is at the root. [https://usds.gov/](https://usds.gov/), this leads to url building confusion and different approaches over time.

- Another issue with links are offsite links should open in new tabs and have a style applied that shows that they are offsite

> image

- Part of checking requires setting the base url correctly for produciton vs staging vs localdev so the plugin KNOWS what's external.

- A handy tool to check links is https://validator.w3.org/checklink
- Need to verify that ON-SITE links still work everywhere.

### CODE REVIEW

File: _includes/new-page.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ The first paragraph above has an example of a markdown link to a different site.
 For links within USDS, there is a special URL format that is needed to make things work across development, staging and on the live site. The brackets-followed-by-parentheses still applies. Inside those parentheses, the first part of the URL must be double curly braces containing `site.baseurl`
 
 - Yes: [this link]({{ site.baseurl }}/mission) will always be right.
```
Comment (TomNUSDS) [inline _includes/new-page.md:37]: I need to fix this documentation. `[this link](mission)` is the correct approach with this change.

**Comment by drew-usds** [comment]
@TomNUSDS The GitHub Pages flavor of Jekyll we're using doesn't allow all external plugins. 

From [plugins section of documentation](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#plugins)

>GitHub Pages cannot build sites using unsupported plugins. If you want to use unsupported plugins, generate your site locally and then push your site's static files to GitHub.

The list of what is/isn't supported is kind of opaque, in that it doesn't say "YES you can use this" in plain language, but seems to be just what is in this [dependency versions](https://pages.github.com/versions/) list.

We could shift how we build the site with GH Actions to use our own plugins (and the most current version of Jekyll too), but that's a different kinda PR!

**Comment by TomNUSDS** [comment]
> @TomNUSDS The GitHub Pages flavor of Jekyll we're using doesn't allow all external plugins.
> 
> From [plugins section of documentation](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#plugins)
> 
> > GitHub Pages cannot build sites using unsupported plugins. If you want to use unsupported plugins, generate your site locally and then push your site's static files to GitHub.
> 
> The list of what is/isn't supported is kind of opaque, in that it doesn't say "YES you can use this" in plain language, but seems to be just what is in this [dependency versions](https://pages.github.com/versions/) list.
> 

Yeah, I was going to try it on staging, but you're probably right.

> We could shift how we build the site with GH Actions to use our own plugins (and the most current version of Jekyll too), but that's a different kinda PR!

Changing the whole build system probably means changing the staging system and all the documentation.

I could probably do it in a couple of lines of javascript? Let me experiment.

**Comment by TomNUSDS** [comment]
@drew-usds I have it working as javascript. One issue I hit (it was also in the prior PR) was this:

image

I could detect buttons and not put the external class on them... but it kind of IS an external link, so technically it should be there.

I'm going to bundle this change up as a PR so you can play with it.

**Comment by drew-usds** [comment]
@TomNUSDS sorry finally looking at this one!

I just see the  `gem 'jekyll-relative-links'` got removed but `gem 'jekyll-target-blank'` is still in the most recent commit.

Were you wanting to test to see if the `target-blank` one works on staging or test the JS version of the external links? I'm totally game for seeing if GH Pages allows the `target-blank` one on staging.

**Comment by TomNUSDS** [comment]
No it’s a mistake. I’ll remove it.
Sent from my iPhone

On Apr 16, 2024, at 8:32 PM, Drew Gardner ***@***.***> wrote:

﻿

@TomNUSDS<https://github.com/TomNUSDS> sorry finally looking at this one!

I just see the gem 'jekyll-relative-links' got removed but gem 'jekyll-target-blank' is still in the most recent commit.

Were you wanting to test to see if the target-blank one works on staging or test the JS version of the external links? I'm totally game for seeing if GH Pages allows the target-blank one on staging.

**Comment by TomNUSDS** [comment]
Closing this PR in favor of https://github.com/usds/website/pull/953

**Pull request closed without merging**